config/interpolate_walk.go

```diff
@@ -150,13 +165,24 @@ func (w *interpolationWalker) Primitive(v reflect.Value) error {
 		// set if it is computed. This behavior is different if we're
 		// splitting (in a SliceElem) or not.
 		remove := false
-		if w.loc == reflectwalk.SliceElem && IsStringList(replaceVal) {
-			parts := StringList(replaceVal).Slice()
-			for _, p := range parts {
-				if p == UnknownVariableValue {
+		if w.loc == reflectwalk.SliceElem {
+			switch typedReplaceVal := replaceVal.(type) {
+			case string:
+				if typedReplaceVal == UnknownVariableValue {
 					remove = true
 					break
 				}
+			case ast.Variable:
+				if hasUnknownValue(typedReplaceVal) {
+					remove = true
+				}
+			case []string:
+				for _, p := range typedReplaceVal {
+					if p == UnknownVariableValue {
+						remove = true
+						break
+					}
```

Comment:
For now this replicates the existing behaviour - in the case of specific targeting this will need to change though. I didn't look at that here beyond verifying that it's no _harder_ to fix than it was before.